### ISSUE TITLE
colcon test failing for fogros2 

### ISSUE THREAD

**Issue by KeplerC** [BerkeleyAutomation/FogROS2#63]
**Describe the bug**
FogROS2 (only fogros2, not fogros2_examples) does not pass the `colcon test`. Needs further investigations on why it does not. 

**To Reproduce**
```bash
cd ros_ws
rm -rf install
colcon build 
colcon test
```

**Expected behavior**
Should pass the `colcon test` without failures 

** logs ** 
the current log is as following: 
```
gdpmobile11@gdpmobile11-NUC7PJYH:~/ros_ws$ colcon test
Starting >>> fogros2
Starting >>> fogros2_examples
--- stderr: fogros2
=============================== warnings summary ===============================
../../../../.local/lib/python3.8/site-packages/pytest_asyncio/plugin.py:191
  Warning: The 'asyncio_mode' default value will change to 'strict' in future, please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
---
Failed   <<< fogros2 [2.22s, exited with code 2]
--- stderr: fogros2_examples
=============================== warnings summary ===============================
../../../../.local/lib/python3.8/site-packages/pytest_asyncio/plugin.py:191
  Warning: The 'asyncio_mode' default value will change to 'strict' in future, please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.

../../../../.local/lib/python3.8/site-packages/_pytest/nodes.py:633
  Warning: The (fspath: py.path.local) argument to Package is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path

../../../../../../opt/ros/rolling/lib/python3.8/site-packages/launch_testing/pytest/hooks.py:179
../../../../../../opt/ros/rolling/lib/python3.8/site-packages/launch_testing/pytest/hooks.py:179
../../../../../../opt/ros/rolling/lib/python3.8/site-packages/launch_testing/pytest/hooks.py:179
  Warning: The (fspath: py.path.local) argument to Module is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path

../../../../../../usr/lib/python3/dist-packages/pydocstyle/config.py:6
  Warning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
---
Finished <<< fogros2_examples [3.32s]	[ with test failures ]

Summary: 1 package finished [5.41s]
  1 package failed: fogros2
  2 packages had stderr output: fogros2 fogros2_examples
  1 package had test failures: fogros2_examples
```

**Comment by KeplerC** [comment]
By running `colcon test --event-handlers console_cohesion+`, it shows: 
```
________________________ ERROR collecting test session _________________________
/usr/lib/python3/dist-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/usr/lib/python3/dist-packages/pluggy/manager.py:92: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/usr/lib/python3/dist-packages/pluggy/manager.py:83: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
../../../../.local/lib/python3.8/site-packages/_pytest/python.py:206: in pytest_collect_file
    module: Module = ihook.pytest_pycollect_makemodule(
../../../../.local/lib/python3.8/site-packages/_pytest/config/compat.py:67: in fixed_hook
    return hook(**kw)
/usr/lib/python3/dist-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/usr/lib/python3/dist-packages/pluggy/manager.py:92: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/usr/lib/python3/dist-packages/pluggy/manager.py:83: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
/opt/ros/rolling/lib/python3.8/site-packages/launch_testing/pytest/hooks.py:161: in pytest_pycollect_makemodule
    entrypoint = find_launch_test_entrypoint(path)
/opt/ros/rolling/lib/python3.8/site-packages/launch_testing/pytest/hooks.py:155: in find_launch_test_entrypoint
    return getattr(path.pyimport(), 'generate_test_description', None)
../../../../.local/lib/python3.8/site-packages/py/_path/local.py:704: in pyimport
    __import__(modname)
fogros2/__init__.py:13: in <module>
    from .aws import AWS
fogros2/aws.py:31: in <module>
    class CloudInstance(abc.ABC):
fogros2/aws.py:34: in CloudInstance
    ros_workspace=os.path.dirname(os.getenv("COLCON_PREFIX_PATH")),
/usr/lib/python3.8/posixpath.py:152: in dirname
    p = os.fspath(p)
E   TypeError: expected str, bytes or os.PathLike object, not NoneType
```
likely to caused by `COLCON_PREFIX_PATH`

**Comment by KeplerC** [comment]
Fixed by [05f07ac](https://github.com/BerkeleyAutomation/FogROS2/commit/05f07aca507371e2a9eca7d2640f555e0596acdc)
